src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -1,53 +1,84 @@
 package frc.robot.subsystems;
 
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.kauailabs.navx.frc.AHRS;
 import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkLowLevel.MotorType;
 
+import edu.wpi.first.math.estimator.DifferentialDrivePoseEstimator;
 import edu.wpi.first.math.geometry.Pose2d;
 import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.simulation.DifferentialDrivetrainSim;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
+import frc.robot.constants.Constants;
+import frc.robot.constants.DriveConstants;
 
 public class Drivetrain extends SubsystemBase {
   
-  private CANSparkMax leftMotor1;
-  private CANSparkMax leftMotor2;
-  private CANSparkMax rightMotor1;
-  private CANSparkMax rightMotor2;
+  // private TalonFX leftMotor1;
+  // private TalonFX leftMotor2;
+  // private TalonFX rightMotor1;
+  // private TalonFX rightMotor2;
+  CANSparkMax leftMotor1;
+  CANSparkMax rightMotor1;
+
+  DifferentialDrivetrainSim cakeOrFake;
 
   // TODO 2.2.1: Create gyro (AHRS)
 
   // TODO 2.2.3: Create DifferentialDriveKinematics
-
+  AHRS abortionLaws = new   AHRS(SPI.Port.kMXP);
   // TODO 2.2.4: Create DifferentialDrivePoseEstimator
 
   // TODO 6.1.5: Create Feedforward and PIDs
 
 
   public Drivetrain() {
     // TODO 1.1.2: Initialize motors
-
-    // TODO 1.1.3: Set motors to brake mode
-
+    if (RobotBase.isReal()) {
+      leftMotor1 = new CANSparkMax(DriveConstants.LEFT_MOTOR_1_ID, MotorType.kBrushless);
+      rightMotor1 = new CANSparkMax(DriveConstants.RIGHT_MOTOR_1_ID, MotorType.kBrushless);
+      // TODO 1.1.3: Set motors to brake mode
+      leftMotor1.setIdleMode(IdleMode.kBrake);
+      rightMotor1.setIdleMode(IdleMode.kBrake);
+    }
     // TODO 1.1.4: Make motor2s follow motor1s
-
+    
     // TODO 1.2.4: Invert motors if necessary
 
     // TODO 2.1.1: Create DifferentialDriveSim if the robot isn't real
-
+    else {
+        cakeOrFake = new DifferentialDrivetrainSim(
+        DriveConstants.DRIVETRAIN_PLANT,
+        DriveConstants.MOTOR, 
+        DriveConstants.GEAR_RATIO, 
+        DriveConstants.TRACK_WIDTH, 
+        DriveConstants.WHEEL_DIAMETER/2, 
+        DriveConstants.MEASUREMENT_STD_DEVS
+      );
+    }
   }
 
   /**
    * This will be called every 20ms, or 50 times per second
    */
   @Override
-  public void periodic(){
+  public void periodic() {
     // TODO 2.2.5: Update odometry
 
     // TODO 1.2.2: Call tankDrive()
-
+    tankDrive(Robot.driver.getLeftTranslation() * .25, Robot.driver.getRightTranslation() * .25);
```

Comment:
This works, but you should multiply by .25 in tankDrive() instead of here so it works in later activites.